server/v2/helpers/tripHelpers/tripsRelationships.js

```diff
@@ -0,0 +1,20 @@
+import tripModel from '../../models/tripsModel';
+import Response from '../responseHelper';
+
+export default async (req, res, next) => {
+  const { body } = req;
+  const busses = await tripModel.findBus(body.bus_license_number);
+  let len = 0;
+  if (busses.length > 0) {
+    while (len < busses.length) {
```

Comment:
I loop through the busses that were retrieved from the database to determine if they have the same trip date as the trip date in the request body. If the dates are the same, Then I determine that the bus is booked.